crates/pixi_glob/src/glob_set.rs

```diff
@@ -140,4 +173,84 @@ mod tests {
+        // ../pixi.toml or ../sources/*.toml, we are able to
+        // distinguish between glob and just a semantic path.
+        let temp_dir = tempdir().unwrap();
+        let root_path = temp_dir.path();
+
+        let temp_path_as_root = temp_dir.path().join("somewhere_inside");
+        fs_err::create_dir(&temp_path_as_root).unwrap();
+
+        // Create files and directories
+        File::create(root_path.join("include1.txt")).unwrap();
+        File::create(root_path.join("include2.log")).unwrap();
+        File::create(root_path.join("exclude.txt")).unwrap();
+        fs_err::create_dir(root_path.join("subdir")).unwrap();
+        File::create(root_path.join("subdir/some_inner_source.cpp")).unwrap();
+
+        // Test globs: include all .txt but exclude exclude.txt
+        let filter_globs = GlobSet::create(vec!["../**/*.cpp", "!exclude.txt"]).unwrap();
+
+        // Filter directory and get results as strings
+        let mut filtered_files: Vec<_> = filter_globs
+            // pretend that we are in the workspace folder
+            // and our recipe yaml is living inside some folder
+            // that will point outside
+            .filter_directory(&temp_path_as_root)
+            .collect::<Result<Vec<_>, _>>()
+            .unwrap()
+            .into_iter()
+            .map(|p| {
+                p.path()
+                    .strip_prefix(&temp_path_as_root)
+                    .unwrap()
+                    .to_path_buf()
+            })
+            .collect();
+
+        // Assert the expected files are present
+        filtered_files.sort();
+
+        let expected = vec![
+            "../subdir/some_inner_source.cpp"
+                .parse::<PathBuf>()
+                .unwrap(),
+        ];
+        assert_eq!(filtered_files, expected);
+    }
+
+    #[test]
+    fn test_filters_with_just_a_file_glob() {
+        let temp_dir = tempdir().unwrap();
+        let root_path = temp_dir.path();
+
+        // Create files and directories
+        File::create(root_path.join("pixi.toml")).unwrap();
+
+        // Test globs: include all .txt but exclude exclude.txt
+        let filter_globs = GlobSet::create(vec!["pixi.toml", "!exclude.txt"]).unwrap();
+
+        // Filter directory and get results as strings
+        let mut filtered_files: Vec<_> = filter_globs
+            // pretend that we are in the workspace folder
+            // and our recipe yaml is living inside some folder
+            // that will point outside
+            .filter_directory(root_path)
+            .collect::<Result<Vec<_>, _>>()
+            .unwrap()
+            .into_iter()
+            .map(|p| p.path().strip_prefix(root_path).unwrap().to_path_buf())
+            .collect();
+
+        // Assert the expected files are present
+        filtered_files.sort();
+
+        let expected = vec!["pixi.toml".parse::<PathBuf>().unwrap()];
+        assert_eq!(filtered_files, expected);
+    }
```

Comment:
This test doesn't have a glob with `..` in front.
Why did you add it?

Reply:
I wanted to test this edge case:
https://github.com/prefix-dev/pixi/pull/3812#discussion_r2102020717

Reply:
Okay!